File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        
        String list = "";
        
        for(int i=1; i<food.length; i++){
            int div = food[i]/2;
            String str = Integer.toString(i);
            list = list+str.repeat(div);
        }
        
        StringBuilder sb = new StringBuilder(list);
        
        String answer = list + "0" +sb.reverse().toString();
        
        return answer;
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    
    var list = "";
    
    for(let i=1; i<food.length; i++){
        var div = food[i]/2;
        var str = String(i);
        list += str.repeat(div);
    }
    
    var answer = list + "0"+ [...list].reverse().join("");
    
    return answer;
}